java/src/main/java/vw/learner/VWBase.java

```diff
@@ -69,12 +71,15 @@ public boolean equals(Object o) {
 
         VWBase vwBase = (VWBase) o;
 
-        return nativePointer == vwBase.nativePointer;
+        if (nativePointer != vwBase.nativePointer) return false;
```

Comment:
Why not `return nativePointer == vwBase.nativePointer && predictionFunctionPointer == vwBase.predictionFunctionPointer;`? Readability?